code-experiments/src/c_linear.c

```diff
@@ -310,22 +317,31 @@ static coco_problem_t *c_linear_single_cons_bbob_problem_allocate(const size_t f
                                 + 50000 * constraint_number);
   srand(seed_cons_i);
   
+  /* The constraints gradients are scaled with random numbers
+   * 10**U[0,1] and 10**U_i[0,2], where U[a, b] is uniform in [a,b] 
+   * and only U_i is drawn for each constraint individually. 
+   * The random number 10**U[0,1] is given by the variable 'factor1' 
+   * while the random number 10**U_i[0,2] is calculated below and 
+   * stored as 'factor2'. (The exception is when the number of
+   * constraints is n+1, in which case 'factor2' defines a random
+   * number 10**U_i[0,1])
+   */
+     
+  if (number_of_linear_constraints == dimension + 1)
+    exp2 = rand() / (RAND_MAX / (1 + 1) + 1);
+  else 
+    exp2 = rand() / (RAND_MAX / (2 + 1) + 1);
```

Comment:
It seems that one way for having `U[M,N]` would be:
1) Declaring `exp2` as an `int`
2) `exp2 = (rand() / (RAND_MAX + 1.0)) * (N - M + 1))`

Because of `1.0`, `(rand() / (RAND_MAX + 1.0))` is a double between 0.0 (inclusive) and  0.9999 (to four decimal places). Assume that `M = 0`. `(rand() / (RAND_MAX + 1.0)) * (N - M + 1))` will then be between 0.0 and (`N` + 0.9999). Since `exp2` is an `int`, the fraction is discarded and it assumes a value between 0.0 and `N`.